frontend/src/container/LogDetailedView/ContextView/ContextLogRenderer.tsx

```diff
@@ -0,0 +1,119 @@
+import './ContextLogRenderer.styles.scss';
+
+import RawLogView from 'components/Logs/RawLogView';
+import Spinner from 'components/Spinner';
+import ShowButton from 'container/LogsContextList/ShowButton';
+import { ORDERBY_FILTERS } from 'container/QueryBuilder/filters/OrderByFilter/config';
+import { useCallback, useEffect, useState } from 'react';
+import { Virtuoso } from 'react-virtuoso';
+import { ILog } from 'types/api/logs/log';
+import { Query, TagFilter } from 'types/api/queryBuilder/queryBuilderData';
+
+import { useContextLogData } from './useContextLogData';
+
+function ContextLogRenderer({
+	isEdit,
+	query,
+	log,
+	filters,
+}: ContextLogRendererProps): JSX.Element {
+	const [firstLog, setFirstLog] = useState<ILog>(log);
+	const [lastLog, setLastLog] = useState<ILog>(log);
+	const [prevLogPage, setPrevLogPage] = useState<number>(1);
+	const [afterLogPage, setAfterLogPage] = useState<number>(1);
+	const [logs, setLogs] = useState<ILog[]>([log]);
+
+	const {
+		logs: previousLogs,
+		isFetching: isPreviousLogsFetching,
+		handleShowNextLines: handlePreviousLogsShowNextLine,
+	} = useContextLogData({
+		log: firstLog,
+		filters,
+		isEdit,
+		query,
+		order: ORDERBY_FILTERS.ASC,
+		page: prevLogPage,
+		setPage: setPrevLogPage,
+	});
+
+	const {
+		logs: afterLogs,
+		isFetching: isAfterLogsFetching,
+		handleShowNextLines: handleAfterLogsShowNextLine,
+	} = useContextLogData({
+		log: lastLog,
+		filters,
+		isEdit,
+		query,
+		order: ORDERBY_FILTERS.DESC,
+		page: afterLogPage,
+		setPage: setAfterLogPage,
+	});
+
+	useEffect(() => {
+		setLogs((prev) => [...previousLogs, ...prev]);
+		setFirstLog(logs[0]);
```

Comment:
logs state won't be immediately updated here, is this expected to have firstLog according to the last state here ?

Reply:
Change it to `previousLogs[0]`

Reply:
On further investigation, I found out that we don't need to set the firstLog or the lastLog, as the dependency to fetch new data is on active-log and offset. The offset will change every time we send a new request to fetch more data. So removed it. 

https://github.com/SigNoz/signoz/assets/26198062/eeb0d2d3-096c-4101-8895-5b0647cc377e